src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/PagedBufferedTextWriter.cs

```diff
@@ -83,11 +83,21 @@ public override void Write(char value)
 
         public override void Write(char[] buffer)
         {
+            if (buffer == null)
+            {
+                return;
+            }
+
             Write(buffer, 0, buffer.Length);
         }
 
         public override void Write(char[] buffer, int index, int count)
         {
+            if (buffer == null)
+            {
+                throw new ArgumentNullException(nameof(buffer));
```

Comment:
@ryanbrandenburg your changes make this class act like almost every other `TextWriter`. That fixes inconsistencies as well as the `Raw()` problem. Having this method do anything other than `throw` for `buffer == null` creates inconsistencies and likely doesn't affect any scenario except user code calling this method directly. Even then, it just makes the behaviour of that code vary depending on exactly where in the `.cshtml` file it appears.